cinema/tests/test_movies.py

```diff
@@ -0,0 +1,160 @@
+        self.assertEqual(res.status_code, status.HTTP_401_UNAUTHORIZED)
+
+        res = self.client.post(reverse("cinema:movie-list"))
+        self.assertEqual(res.status_code, status.HTTP_401_UNAUTHORIZED)
+
+        res = self.client.get(reverse("cinema:movie-detail", args=[1]))
+        self.assertEqual(res.status_code, status.HTTP_401_UNAUTHORIZED)
+
+        res = self.client.get(reverse("cinema:movie-upload-image", args=[1]))
+        self.assertEqual(res.status_code, status.HTTP_401_UNAUTHORIZED)
+
+        res = self.client.delete(
+            reverse(
+                "cinema:movie-detail",
+                args=[
+                    1,
+                ],
+            )
+        )
+        self.assertEqual(res.status_code, status.HTTP_401_UNAUTHORIZED)
+
+    def test_user_access(self):
+        self.client.force_authenticate(self.user)
+
+        res = self.client.get(reverse("cinema:movie-upload-image", args=[1]))
+        self.assertEqual(res.status_code, status.HTTP_403_FORBIDDEN)
+
+        res = self.client.delete(
+            reverse(
+                "cinema:movie-detail",
+                args=[
+                    1,
+                ],
+            )
+        )
+        self.assertEqual(res.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_admin_access(self):
+        self.client.force_authenticate(self.superuser)
+
+        res = self.client.get(reverse("cinema:movie-list"))
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+
+        res = self.client.get(reverse("cinema:movie-detail", args=[1]))
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+
+        res = self.client.delete(
+            reverse(
+                "cinema:movie-detail",
+                args=[
+                    1,
+                ],
+            )
+        )
+        self.assertEqual(res.status_code, status.HTTP_405_METHOD_NOT_ALLOWED)
+
+    def test_movie_list(self):
+        self.client.force_authenticate(self.user)
+        res = self.client.get(reverse("cinema:movie-list"))
+
+        serializer = MovieListSerializer(Movie.objects.all(), many=True)
+
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+        self.assertEqual(res.data, serializer.data)
+
+    def test_movie_list_with_title_search(self):
+        self.client.force_authenticate(self.user)
+        res = self.client.get(reverse("cinema:movie-list"), {"title": "Sample1"})
+
+        serializer_movies = MovieListSerializer(
+            Movie.objects.filter(title__icontains="Sample1"), many=True
+        )
+
+        self.assertEqual(res.data, serializer_movies.data)
+
+    def test_movie_list_with_genres_search(self):
+        self.client.force_authenticate(self.user)
+        res = self.client.get(reverse("cinema:movie-list"), {"genres": ["101", "102"]})
+
+        serializer_movies = MovieListSerializer(
+            Movie.objects.filter(genres__name__icontains=("101", "102")), many=True
```

Comment:
The filter condition `genres__name__icontains=("101", "102")` might not work as intended. Consider using `genres__name__in=["101", "102"]` to filter by multiple genre names.